lib/rspec/rails/matchers/active_job.rb

```diff
@@ -30,6 +31,11 @@ def on_queue(queue)
             self
           end
 
+          def with_priority(priority)
```

Comment:
Personally I would call this `priority` or `at_priority` to disambigute from `with`